### ISSUE TITLE
Emscripten calling fetch from pthread seems to hang forever.

### ISSUE THREAD

**Issue by FrozenSource** [emscripten-core/emscripten#20500]
**Version of emscripten/emsdk:**
emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) 3.1.46-git
clang version 18.0.0 (https://github.com/llvm/llvm-project.git 75501f53624de92aafce2f1da698b249a7293dc7)
Target: wasm32-unknown-emscripten
Thread model: posix
InstalledDir: /opt/homebrew/Cellar/emscripten/3.1.46/libexec/llvm/bin

Hi it might be that I am just missing something completely. But I have this snippet of code here and no matter wat EMSCRIPTEN_FETCH_* attribute I slap on there the fetch keeps blocking with no sight of progress. Even if the minimal case with callbacks just seems to block forever. No traffic in the network tab no breakpoint in the JS start_fetch or the others. Quite simply one hell to debug.

Is this something I am doing wrong or is it emscripten doing something that I don't see. request_t and response_t are just wrappers over the emscripten_fetch_t*.

Off course compiled with -sUSE_PTHREADS=1 and -sPTHREAD_POOL_SIZE=navigator.hardwareConcurrency. Have to mention that this is with MEMORY_GROWTH but up until now this has not been a performance problem at all.
```
    inline std::optional<response_t> _send(request_t& _req) {
        auto& req = *(_request_t*) _req.handle();

        std::string url = (req.url.Protocol.empty() && req.url.Host.empty()) ? req.url.createPath(false) : req.url.toString();
        req.attr.userData = &_req;
        req.attr.attributes = EMSCRIPTEN_FETCH_TO_MEMORY | EMSCRIPTEN_FETCH_SYNCHRONOUS;
        // Use a promise and a future to communicate data between threads
        std::promise<emscripten_fetch_t*> promise;
        std::future<emscripten_fetch_t*> future = promise.get_future();

        // Create a worker thread
        std::thread workerThread([&]() {
            auto fetch = emscripten_fetch(&req.attr, url.c_str());
            promise.set_value(fetch);  // Set the data for the main thread
        });

        auto fetch = future.get();
        workerThread.join();

        return new _response_t { fetch };
    }
```

Thanks in advance,

**Comment by FrozenSource** [comment]
FYI: This seems to go OK in edge, but only seems to occur in Firefox (haven't been able to test Safari yet).

I got it working one time by coincidence as running the script obviously completely breaks the browser. After some time you get the following:
![image](https://github.com/emscripten-core/emscripten/assets/14978036/a6f4b200-64f1-45cd-9443-114c3d986682)

When I pressed debug script it somehow succeeded, can't reproduce that anymore though as it once again just hangs.
Maybe somebody who is more experienced could debug Firefox itself or whatever cause I can't find any other info.

**Comment by sbc100** [comment]
Maybe take a look at the test_fetch_sync_xhr test in the browser test suite?

This test passed for me when I run `./test/runner browser.test_fetch_sync_xhr`

**Comment by sbc100** [comment]
Is `req.attr` correctly initialized?  e.g. with `emscripten_fetch_attr_init(...);`?

**Comment by FrozenSource** [comment]
yes I use the init function. So as far as I know no memory problem. I have reproduced it for you in a minimal sample see the attached zip.
Funny thing is that for the first 

time I see the blocking main thread warning and not in production... but I do need it to be blocking in this case.

Hope you can help me out :D

[test.zip](https://github.com/emscripten-core/emscripten/files/13161875/test.zip)

**Comment by sbc100** [comment]
I ran the code in test.zip and it appeared to work just fine.  Perhaps this issue has been fixed in the intervening year.  Can you confirm?
